docs/source/tutorials/datasets.rst

```diff
@@ -172,6 +244,104 @@ Here is how messages would be formatted using the :class:`~torchtune.data.Llama2
 Note that the system message is now incorporated in the user message. If you create custom ChatFormats
 you can also add more advanced behavior.
 
+.. code-block:: python
+
+    from torchtune.datasets import chat_dataset
+    from torchtune.data import ChatFormat
+
+    class CustomChatFormat(ChatFormat):
+        # Define templates for system, user, assistant messages
+        # as strings with {} as placeholders for message content
+        system = ...
+        user = ...
+        assistant = ...
+
+        # Implement this method
+        @classmethod
+        def format(
+            cls,
+            sample: List[Message],
+        ) -> List[Message]:
+            ...
+
+    # Load in tokenizer
+    tokenizer = ...
+    dataset = chat_dataset(
+        tokenizer=tokenizer,
+        source="my/dataset/path",
+        conversation_style="openai",
+        chat_format="import.path.to.CustomChatFormat",
+    )
+
+.. code-block:: yaml
+
+    # YAML config
+    dataset:
+      _component_: torchtune.datasets.chat_dataset
+      source: my/dataset/path
+      conversation_style: openai
+      chat_format: import.path.to.CustomChatFormat
+
+.. code-block:: bash
+
+    # Command line
+    tune run full_finetune_single_device --config llama3/8B_full_single_device dataset=torchtune.datasets.chat_dataset dataset.source=my/dataset/path dataset.conversation_style=openai dataset.chat_format=import.path.to.CustomChatFormat
+
+
+Multiple in-memory datasets
+---------------------------
+
+It is also possible to train on multiple datasets by combining them into a single :class:`~torchtune.datasets.ConcatDataset`.
```

Comment:
Is ConcatDataset inferred here somewhere? Does the dataset init know what to do when it sees a list? If so we should make that clear as doing a ConcatDataset in code would look different.

Reply:
The recipes were updated to handle init-ing a list. I can add that code snippet here, but frankly we're in the process of migrating to IterableDatasets and `ConcatDataset` may change significantly

Reply:
makes sense, I'll remove mention of it but keep the yaml example